### DISCUSSION TITLE
Should the Chem.FindMolChiralCenters function be sensitive to atom-map numbers on molecules?

### DISCUSSION THREAD

**Discussion by skystreet8** [rdkit/rdkit#6931] [Q&A] [answered]
Dear all,

I'm using the Chem.FindMolChiralCenters function to identify all possible chiral atoms for a bunch of molecules. Since I want to visualize the positions of these chiral atoms, I manually set the atom-map numbers of the molecule's atoms to equal their atom indexes + 1. I find that for some achiral molecules like tert-butyl alcohol, if the atom-map numbers around the center carbon atom(see below) are set differently then the atom would be recognized as an unspecified chiral atom. This wouldn't happen if the atom-map numbers were not set.

My RDKit version is 2023.03.3, OS is Win 11. Here is my code:
```python3
from rdkit import Chem
from rdkit.Chem import Draw
from IPython.display import display_png
from PIL import Image
from io import BytesIO
```

```python3
smi = 'CC(C)(C)O'
m = Chem.MolFromSmiles(smi)
for a in m.GetAtoms():
    a.SetAtomMapNum(a.GetIdx() + 1)  # Set the atom-map numbers leads to my question
chiral_centers = Chem.FindMolChiralCenters(m, force=True, includeUnassigned=True, useLegacyImplementation=True, includeCIP=False)
print(chiral_centers)  # An center carbon atom would be assigned with unspecified chirality, shown in yellow highlight
drawer = Draw.rdMolDraw2D.MolDraw2DCairo(500, 300)
dopt = drawer.drawOptions()
dopt.useBWAtomPalette()
drawer.DrawMolecule(m, highlightAtoms=[t[0] for t in chiral_centers], highlightBonds=None, highlightAtomColors={t[0]: (1,1,0) for t in chiral_centers},
                    highlightAtomRadii={t[0]: 0.5 for t in chiral_centers})
drawer.FinishDrawing()
png = drawer.GetDrawingText()
png = Image.open(BytesIO(png))
display_png(png)
```

```python3
[(1, '?')]
```

![image](https://github.com/rdkit/rdkit/assets/108856565/1a7e2202-f3d5-45ae-abea-62066187c393)
```python3
smi = 'CC(C)(C)O'
m = Chem.MolFromSmiles(smi)
# for a in m.GetAtoms():
    # a.SetAtomMapNum(a.GetIdx() + 1)
chiral_centers = Chem.FindMolChiralCenters(m, force=True, includeUnassigned=True, useLegacyImplementation=False, includeCIP=False)
print(chiral_centers)  # Works as normal
drawer = Draw.rdMolDraw2D.MolDraw2DCairo(500, 300)
dopt = drawer.drawOptions()
dopt.useBWAtomPalette()
drawer.DrawMolecule(m, highlightAtoms=[t[0] for t in chiral_centers], highlightBonds=None, highlightAtomColors={t[0]: (1,1,0) for t in chiral_centers},
                    highlightAtomRadii={t[0]: 0.5 for t in chiral_centers})
drawer.FinishDrawing()
png = drawer.GetDrawingText()
png = Image.open(BytesIO(png))
display_png(png)
```
```python3
[]
```
![image](https://github.com/rdkit/rdkit/assets/108856565/c83a38f7-42f8-44c2-a82e-7ae0700f02a4)
Should this be the expected behavior of the function? Thanks a lot!

**Comment by bp-kelley** [comment, marked as answer]
I think this might be an artifact of the new chirality detection which uses the smiles invariants to help detect chiral centers.

I personally think this is a bug or at least, should be optional.  I'll follow up with greg.

**Reply by bp-kelley** [reply]
We are putting it in the queue to be fixed.

**Reply by greglandrum** [reply]
@skystreet8 can you please click the "Create issue from discussion" link on this page so that you get credit for the issue in the next RDKit release notes?

**Reply by skystreet8** [reply]
@greglandrum @bp-kelley Thank you for the reply. Hope this would get fixed soon.
